front/admin_ui/app.py

```diff
@@ -0,0 +1,121 @@
+PROD_DSS_ENDPOINT = os.environ.get("PROD_DSS_ENDPOINT", "https://datasets-server.huggingface.co")
+DEV_DSS_ENDPOINT = os.environ.get("DEV_DSS_ENDPOINT", "http://localhost:8100")
+ADMIN_HF_ORGANIZATION = os.environ.get("ADMIN_HF_ORGANIZATION", "huggingface")
+
+DSS_ENDPOINT = DEV_DSS_ENDPOINT if DEV else PROD_DSS_ENDPOINT
+
+pending_jobs_df = None
+
+def healthcheck():
+    response = requests.head(f"{DSS_ENDPOINT}/admin/pending-jobs", timeout=10)
+    if response.status_code == 401:
+        return f"*Connected to {DSS_ENDPOINT}*"
+
+    else:
+        return f"❌ Failed to connect to {DSS_ENDPOINT} (error {response.status_code})"
+
+
+
+with gr.Blocks() as demo:
+    gr.Markdown(" ## Datasets-server admin page")
+    gr.Markdown(healthcheck)
+
+    with gr.Row() as auth_page:
+        with gr.Column():
+            auth_title = gr.Markdown("Enter your token ([settings](https://huggingface.co/settings/tokens)):")
+            token_box = gr.Textbox(label="token", placeholder="hf_xxx", type="password")
+            auth_error = gr.Markdown("", visible=False)
+    
+    with gr.Row(visible=False) as main_page:
+        with gr.Column():
+            welcome_title = gr.Markdown("")
+            with gr.Tab("View pending jobs"):
+                fetch_pending_jobs_button = gr.Button("Fetch pending jobs")
+                gr.Markdown("### Pending jobs summary")
+                pending_jobs_summary_table = gr.DataFrame(pd.DataFrame({"Jobs": [], "Waiting": [], "Started": []}))
+                gr.Markdown("### Most recent")
+                recent_pending_jobs_table = gr.DataFrame()
+                gr.Markdown("### Query the pending jobs table")
+                pending_jobs_query = gr.Textbox(
+                    label="Query pending_jobs_df",
+                    placeholder="SELECT * FROM pending_jobs_df WHERE dataset LIKE 'allenai/c4",
+                    value="SELECT * FROM pending_jobs_df WHERE dataset LIKE 'allenai/c4'",
+                    lines=3,
+                )
+                query_pending_jobs_button = gr.Button("Run")
+                pending_jobs_query_result_df = gr.DataFrame()
+
+    def auth(token):
+        if not token:
+            return {auth_error: gr.update(value="", visible=False)}
+        try:
+            user = hfh.whoami(token=token)
+        except requests.HTTPError as err:
+            return {auth_error: gr.update(value=f"❌ Error ({err})", visible=True)}
+        orgs = [org["name"] for org in user["orgs"]]
+        if ADMIN_HF_ORGANIZATION in orgs:
+            return {
+                auth_page: gr.update(visible=False),
+                welcome_title: gr.update(value=f"### Welcome {user['name']}"),
+                main_page: gr.update(visible=True)
+            }
+        else:
+            return f"❌ Unauthorized (user '{user['name']} is not a member of '{ADMIN_HF_ORGANIZATION}')"
+
+    def view_jobs(token):
+        global pending_jobs_df
```

Comment:
I'm not sure if this will work for concurrent users if deployed on spaces. The same df will be shared by everyone. 

Might be best to save this in an invisible `gr.Dataframe` component? That will be isolated per session.

Reply:
It's ok if it's shared by everyone for now.

I tried to store it in a `gr.Dataframe` though but it crashed. The dataframe is 40MB and has string and timestamp types that can be null.

Reply:
More specifically I tried to store the pandas dataframe it in a gr.Dataframe with visible=False, and it caused my server to hang